cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -142,10 +139,15 @@ public void resumeWorkflowRun(HttpRequest request, HttpResponder responder,
   /**
    * Returns the {@link ProgramController} for the given workflow program.
    */
-  private ProgramController getProgramController(String namespaceId, String appId,
+  private ProgramController getProgramController(String namespace, String appName,
                                                  String workflowName, String runId) throws NotFoundException {
-    ApplicationId applicationId = new ApplicationId(namespaceId, appId, getLatestAppVersion(namespaceId, appId));
-    ProgramId id = new ProgramId(applicationId, ProgramType.WORKFLOW, workflowName);
+    NamespaceId namespaceId = Ids.namespace(namespace);
+    ApplicationMeta appMeta = store.getLatest(namespaceId, appName);
+    if (appMeta == null || appMeta.getSpec() == null) {
+      throw new NotFoundException(namespaceId.app(appName));
+    }
+    ApplicationId appId =  namespaceId.app(appName, appMeta.getSpec().getAppVersion());
+    ProgramId id = appId.workflow(workflowName);
     ProgramRuntimeService.RuntimeInfo runtimeInfo = runtimeService.list(id).get(RunIds.fromString(runId));
```

Comment:
@chtyim should we move the above logic to runtimeService.list(namespace, appName, workflowName, runId) instead?

Reply:
If we are doing this, the change is pretty big - we can address it in a separate PR. Created https://cdap.atlassian.net/browse/CDAP-19934